kvbc/src/v4blockchain/detail/blockchain.cpp

```diff
@@ -100,4 +144,9 @@ std::optional<BlockId> Blockchain::loadGenesisBlockId() {
   return concordUtils::fromBigEndianBuffer<BlockId>(itr.keyView().data());
 }
 
+void Blockchain::setBlockId(BlockId id) {
+  setLastReachable(id);
+  if (genesis_block_id_ == INVALID_BLOCK_ID) genesis_block_id_ = id;
```

Comment:
Nit: A small query:
We have a addBlock, the genesis_block_id_ should be updated when the addBlock is called for the first time. So this check should happen in addBlock.
We should not give any public function which can set the block id. So setBlockId and setLastReachable should be either made private or should be removed.

Reply:
not sure I got ```We should not give any public function which can set the block id. So setBlockId and setLastReachable should be either made private or should be removed.```

We need the add block method in the v4_blockchain to be able to set the new last reachable ID, because only it knows if the block was successfully added. how should it be done?

Reply:
Actually I was saying that we will only have addBlock function in public in public, which will set and increment the last reachable ID and initialize the genesis block id.

the public functions like setBlockId and setLastReachable should not be available to anyone outside the Blockchain class.

Reply:
the blockchain is a private member of the v4 handler, so no one should be able to call it